<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="input_velocity" default="/sensing/vehicle_velocity_converter/twist_with_covariance"/>
  <arg name="input_objects" default="/perception/object_recognition/objects"/>
  <arg name="validate_calibration_srv" default="validate_calibration"/>
  <arg name="data_path" default="$(env HOME)/autoware_data" description="packages data and artifacts directory path"/>
  <arg name="model_name" default="calibration_status_classifier" description="options: `calibration_status_classifier"/>
  <arg name="model_path" default="$(var data_path)/calibration_status_classifier"/>
  <arg name="model_param_path" default="$(find-pkg-share autoware_calibration_status_classifier)/config/$(var model_name).param.yaml"/>
  <arg name="ml_package_param_path" default="$(var model_path)/ml_package_$(var model_name).param.yaml"/>
  <arg name="build_only" default="false" description="shutdown node after TensorRT engine file is built"/>
  <arg name="use_sim_time" default="true" description="use simulation time"/>

  <arg name="decompressor_param_file" default="$(find-pkg-share autoware_image_transport_decompressor)/config/image_transport_decompressor.param.yaml"/>
  <node pkg="autoware_image_transport_decompressor" exec="image_transport_decompressor_node" name="image8_transport_decompressor_node">
    <remap from="~/input/compressed_image" to="/sensing/camera/camera8/image_raw/compressed"/>
    <remap from="~/output/raw_image" to="/sensing/camera/camera8/image_raw"/>
    <param from="$(var decompressor_param_file)"/>
    <param name="use_sim_time" value="$(var use_sim_time)"/>
  </node>

  <node pkg="autoware_calibration_status_classifier" exec="calibration_status_classifier_node" name="calibration_status_classifier" output="both">
    <param from="$(var model_param_path)" allow_substs="true"/>
    <param from="$(var ml_package_param_path)"/>
    <remap from="~/input/velocity" to="$(var input_velocity)"/>
    <remap from="~/input/objects" to="$(var input_objects)"/>
    <remap from="~/input/validate_calibration_srv" to="$(var validate_calibration_srv)"/>
    <param name="use_sim_time" value="$(var use_sim_time)"/>
    <param name="build_only" value="$(var build_only)"/>
  </node>
</launch>
